scripts/sanity_chk/sanitylib.py

```diff
@@ -2845,136 +2845,152 @@ def process_log(log_file):
+
+            for _, instance in inst.items():
+                handler_time = instance.metrics.get('handler_time', 0)
+                duration += handler_time
+                if full_report:
+                    for k in instance.results.keys():
+                        if instance.results[k] == 'PASS':
+                            passes += 1
+                        elif instance.results[k] == 'BLOCK':
+                            errors += 1
+                        elif instance.results[k] == 'SKIP':
+                            skips += 1
+                        else:
+                            fails += 1
+                else:
+                    if instance.status in ["failed", "timeout"]:
+                        if instance.reason in ['build_error', 'handler_crash']:
+                            errors += 1
                         else:
+                            fails += 1
+                    elif instance.status == 'skipped':
+                        skips += 1
+                    else:
+                        passes += 1
+
+            total = (errors + passes + fails + skips)
+            run = p
+            eleTestsuite = None
+
+            # When we re-run the tests, we re-use the results and update only with
+            # the newly run tests.
+            if os.path.exists(filename) and append:
+                eleTestsuite = eleTestsuites.findall(f'testsuite/[@name="{p}"]')[0]
+                eleTestsuite.attrib['failures'] = "%d" % fails
+                eleTestsuite.attrib['errors'] = "%d" % errors
+                eleTestsuite.attrib['skip'] = "%d" % skips
+            else:
+                eleTestsuite = ET.SubElement(eleTestsuites, 'testsuite',
+                                             name=run, time="%f" % duration,
+                                             tests="%d" % (total),
+                                             failures="%d" % fails,
+                                             errors="%d" % (errors), skip="%s" % (skips))
+
+            for _, instance in inst.items():
+                if full_report:
+                    tname = os.path.basename(instance.testcase.name)
+                else:
+                    tname = instance.testcase.name
+                # remove testcases that are being re-run from exiting reports
+                if append:
+                    for tc in eleTestsuite.findall('testcase'):
+                        if tc.get('name') == "%s" % (tname):
+                            eleTestsuite.remove(tc)
+
+                handler_time = instance.metrics.get('handler_time', 0)
+
+                if full_report:
+                    for k in instance.results.keys():
+
+                        classname = ".".join(tname.split(".")[:2])
+                        eleTestcase = ET.SubElement(
+                            eleTestsuite, 'testcase',
+                            classname=classname,
+                            name="%s" % (k), time="%f" % handler_time)
```

Comment:
handler_time is use the log the processing time, and we will ask testrails to improve proformance if not good?

Reply:
not sure what you mean, this has nothing to do with testrail. This is the time it took to run the test.